shot_debug_manager.py

```diff
@@ -63,18 +88,48 @@ def set_shot_type(self, type: str):
         self.shottype = type
 
 
+def add_logging_handler(handler: logging.Handler):
```

Comment:
You could add it once and then call something like handle_log on the debug manager. That way you don't have to do the cleanup